src/screens/TaskDetailScreen.tsx

```diff
@@ -165,140 +213,162 @@ const TaskDetailScreen: React.FC<TaskDetailScreenProps> = ({ navigation, route }
             style: 'destructive',
             onPress: () => {
-              onDelete(task.id);
-              navigation.goBack();
+              // Fade out animation before navigation
+              Animated.timing(fadeAnim, {
+                toValue: 0,
+                duration: 200,
+                useNativeDriver: true
+              }).start(() => {
+                onDelete(task.id);
+                navigation.goBack();
+              });
             }
           }
         ]
     );
-  };
+  }, [navigation, task.id, onDelete, fadeAnim]);
 
   /**
    * Formats a date string to a more readable format
    * @param dateString - ISO date string to format
    * @returns Formatted date string (e.g., "April 19, 2025")
    */
-  const formatDate = (dateString: string) => {
+  const formatDate = useCallback((dateString: string) => {
     const date = new Date(dateString);
     return date.toLocaleDateString('en-US', {
       year: 'numeric',
       month: 'long',
       day: 'numeric'
     });
-  };
+  }, []);
 
   return (
-      <ScrollView style={styles.container}>
-        {/* Task details card */}
-        <View style={styles.card}>
-          <Text style={styles.title}>{task.title}</Text>
+      <>
+        <StatusBar barStyle="dark-content" />
+        <Animated.ScrollView
+            style={[styles.container, { opacity: fadeAnim }]}
+            showsVerticalScrollIndicator={false}
+        >
+          {/* Task details card */}
+          <View style={styles.card}>
+            <Text style={styles.title}>{task.title}</Text>
 
-          <Text style={styles.sectionTitle}>Description</Text>
-          <Text style={styles.description}>
-            {task.description || 'No description'}
-          </Text>
+            <Text style={styles.sectionTitle}>Description</Text>
+            <Text style={styles.description}>
+              {task.description || 'No description'}
+            </Text>
 
-          <Text style={styles.sectionTitle}>Created Date</Text>
-          <Text style={styles.date}>{formatDate(task.createdAt)}</Text>
+            <Text style={styles.sectionTitle}>Created Date</Text>
+            <Text style={styles.date}>{formatDate(task.createdAt)}</Text>
 
-          <Text style={styles.sectionTitle}>Status</Text>
-          {/* Status badge with styling consistent with task list */}
-          <View style={styles.statusContainer}>
-            <View style={[
-              styles.statusBadge,
-              task.status === 'completed' ? styles.completedBadge : styles.pendingBadge
-            ]}>
-              <Text style={[
-                styles.statusText,
-                task.status === 'completed' ? styles.completedText : styles.pendingText
+            <Text style={styles.sectionTitle}>Status</Text>
+            {/* Status badge with styling consistent with task list */}
+            <View style={styles.statusContainer}>
+              <View style={[
+                styles.statusBadge,
```

Comment:
The render function is quite long. You could pull out:

A <StatusBadge status={task.status} /> component

A generic <ActionButton iconName=… label=… onPress=… color=… />